### PR TITLE
Use ruff

### PR DESCRIPTION
**Pull request by gadomski**
**Description:**
We've moved to ruff in **pystac** et al. Also:
- Upgrade to markdownlint-cli2
- Autoupdate pre-commit hooks
- Consolidate `lint` and `format` scripts into `lint`, since we just use them as wrappers around `pre-commit`

No functional code changes, just formatting changes from **ruff**.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [ ] Documentation has been updated to reflect changes, if applicable.
- [ ] Examples have been updated to reflect changes, if applicable
- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.MD
Comment (jsignell) [inline .github/PULL_REQUEST_TEMPLATE.MD:1]: you can make org-level GH templates if you want to reduce repo-drift

File: .pre-commit-config.yaml
Comment (jsignell) [inline .pre-commit-config.yaml:1]: I'm wondering if we can use a ruff pre-commit hook rather than having two linting paths.

**Review by jsignell — APPROVED** [review]
LGTM

File: .pre-commit-config.yaml
Comment (gadomski) [reply .pre-commit-config.yaml:1]: Sorry, I don't quite understand. In this PR, I've updated the `scripts/lint` to just run the pre-commit hooks.

Comment (jsignell) [reply .pre-commit-config.yaml:1]: Right :facepalm: I got megalinter and ruff confused. Ignore me.

Comment (gadomski) [reply .pre-commit-config.yaml:1]: No prob!

**Pull request merged**